src/SessionManager/ProgramSettings.cc

```diff
@@ -182,6 +283,55 @@ sending messages to the backend).
             files.push_back(relative_path.string());
         }
     }
+
+    // produce JSON for over-ridding system and user configuration;
+    // Options here need to match all options available for system and user settings
+    command_line_settings = json({}); // needs to have empty JSON at least in case of no command line options
+    if (result.count("verbosity")) {
```

Comment:
This is also a lot of boilerplate to check if the keys exist. But I'm not sure if we can easily do anything about this because of the need to cast to different types.

Reply:
The only solution I can think of is something like 

```
vector<string> bool_keys = {"verbosity", "no_log", "log_performance", "log_protocol_messages", ...};

for(auto& key : bool_keys)) {
    if (result.count(key)) {
        command_line_settings[key] = result[key].as<bool>();
    }
}

// Repeat for int and string

```

Reply:
I agree with both of you.